engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -22,6 +22,73 @@ def self.write_cache_file(cache_dir, cache_key)
     FileUtils.touch(File.join(cache_dir, cache_key))
   end
 
+  def self.verify_instance(request, logger, system)
+    return false unless request.headers['X-Instance-Data']
+
+    instance_data = Base64.decode64(request.headers['X-Instance-Data'].to_s)
+    base_product = system.products.find_by(product_type: 'base')
+    return false unless base_product
+
+    # check the cache for the system (20 min)
+    cache_path = File.join(
+      Rails.application.config.repo_cache_dir,
+      InstanceVerification.build_cache_key(request.remote_ip, system.login, base_product_id: base_product.id)
+    )
+    if File.exist?(cache_path)
+      # only update registry cache key
+      InstanceVerification.update_cache(request.remote_ip, system.login, nil, is_byos: system.proxy_byos, registry: true)
```

Comment:
The `is_byos` key was removed with ccb4e6e95 ???

Reply:
The PR was created without the removal of BYOS changes in place

I have merged master branch into this branch so the removal of BYOS param is in place